conan/packager.py

```diff
@@ -84,285 +107,96 @@ def __init__(self, args=None, username=None, channel=None, runner=None,
-                            self.add({"arch": arch,
-                                      "build_type": build_type,
-                                      "compiler": "gcc",
-                                      "compiler.version": gcc_version,
-                                      "compiler.libcxx": "libstdc++"}, {})
-                            if float(gcc_version) > 5:
-                                self.add({"arch": arch,
-                                          "build_type": build_type,
-                                          "compiler": "gcc",
-                                          "compiler.version": gcc_version,
-                                          "compiler.libcxx": "libstdc++11"}, {})
-                        else:
-                            self.add({"arch": arch,
-                                      "build_type": build_type,
-                                      "compiler": "gcc",
-                                      "compiler.version": gcc_version}, {})
-
-    def add(self, settings=None, options=None):
+        builds = []
+        if self._platform_info.system() == "Windows":
+            if self.mingw_configurations:
+                builds = get_mingw_builds(self.mingw_configurations, self.mingw_installer_reference, self.archs)
+            builds.extend(get_visual_builds(self.visual_versions, self.archs, self.visual_runtimes,
+                                            shared_option_name, dll_with_static_runtime, self.vs10_x86_64_enabled))
+        elif self._platform_info.system() == "Linux" or self.use_docker is True:
+            builds = get_linux_gcc_builds(self.gcc_versions, self.archs, shared_option_name, pure_c)
+        elif self._platform_info.system() == "Darwin":
+            builds = get_osx_apple_clang_builds(self.apple_clang_versions, self.archs, shared_option_name, pure_c)
+
+        self.builds.extend(builds)
+
+    def add(self, settings=None, options=None, env_vars=None, build_requires=None):
         settings = settings or {}
         options = options or {}
-        self.builds.append([settings, options])
+        env_vars = env_vars or {}
+        build_requires = build_requires or {}
+        self.builds.append(BuildConf(settings, options, env_vars, build_requires))
 
     def run(self):
-        self._pack()
-        self._upload_packages()
-
-    def pack(self, curpage=1, total_pages=1):
-        self.curpage = int(curpage)
-        self.total_pages = total_pages
-        self._pack()
+        self._pip_install()
+        self.run_builds()
+        self.upload_packages()
+
+    def run_builds(self, curpage=None, total_pages=None):
+        curpage = curpage or int(self.curpage)
+        total_pages = total_pages or int(self.total_pages)
+        self.runner('conan export %s/%s' % (self.username, self.channel))
+
+        # Auto remove conan.conf if exist for perform a fresh detection
+        self.runner("conan info")  # Ensure conan.conf is created
```

Comment:
I think those are not needed anymore. "info" was just for create a conan.conf and the "auto remove" comment, no idea, copy paste not review properly.